### PR TITLE
[PICK & CHOOSE] Enhancements from @pescennius to mitigate #2 and #14

### PR DESCRIPTION
**Pull request by bootmii**
You know @micheletest's PR changing the color of the T to the one on Wikipedia? That inspired me to do a little leg work and find the shade of turquoise the MTA actually uses. They actually use one shade for the line and another for the bullet, but they're only slightly different. I averaged the two and created "the official" shade of turquoise: #05AEDE. I've also made the W train visible by default, just like the Z train is.

I'm also using this PR to suggest some of @pescennius' improvements, such as official service overloading, which mitigates #2 and #14 (although that's short of what I'd like to happen, which is custom services on existing trunks and custom `html` bullets), loading plans from URL, and new default plans (2025 is soon, and only Phase 1 of the SAS has been built, but the W train is back). I advise against Pescennius' name change, and the Vimeo-to-local and SIRR/brown M changes are a matter of taste, but `js/subway-*.js` are ready to go; they have had #21 redone to them.

### CODE REVIEW

**Comment by bootmii** [comment]
Conflicts in subway-main(-headless), (css/)style.css, and game-starters are over the shade of turquoise. @micheletest took her hex triplet from Wikipedia, while I, fearing [citogenesis](en.wikipedia.org/wiki/Citogenesis), went and got the [biggest official image I could](https://new.mta.info/sites/default/files/2020-11/SAS-Graphic-Map-Final-070819.jpg) and used an averaging color picker (because it's a JPEG). What I discovered was that the bullet was slightly darker and more saturated than the line. I averaged the two, because this game always uses the same shade for both.

index(-bk).html's conflict should be resolved in your favor, but it's still worth taking a lot of looks at. GitHub vs. Vimeo hosting for the video is a judgment call, since the Vimeo embed works for me as a desktop Linux Firefox user while @pescennius said it was broken. There are other judgment calls in there too.

**Comment by bootmii** [comment]
subway-files.js has a phantom conflict because I copied @GrahamSH-LLK's contribution by hand. It doesn't matter which side you resolve, just that you resolve it.

**Comment by jpwright** [comment]
Hi, thanks for submitting a PR! As you can probably tell I don't actively work on this project anymore, but I'm happy to merge PRs that look reasonable.

In this case, please resolve all merge conflicts in your branch- and it looks like this PR includes renaming the game and changing the attribution, that should be left for any forks and not merged into this repo.

**Comment by bootmii** [comment]
"Please resolve all merge conflicts in your branch": I'd be happy to, but I'd need write access. Unless there's another way?

**Comment by bootmii** [comment]
Maybe the issue is that I'm a fork of a fork?

**Comment by jpwright** [comment]
> "Please resolve all merge conflicts in your branch": I'd be happy to, but I'd need write access. Unless there's another way?

merge `jpwright:master` into `bootmii:master`. Resolve any merge conflicts on your branch and then commit again.

**Comment by bootmii** [comment]
The conflicts are completely resolved now.

**Comment by bootmii** [comment]
There might be one regression: the city picker doesn't work in the-singularity-labs or mine.

File: package.json
Comment (jpwright) [inline package.json:1]: Please remove for this PR

File: server/views/pages/debug.ejs
Comment (jpwright) [inline server/views/pages/debug.ejs:8]: Revert

File: server/views/partials/head.ejs
Comment (jpwright) [inline server/views/partials/head.ejs:2]: Revert

File: js/subway-main.js
Comment (jpwright) [inline js/subway-main.js:345]: It looks like the effect of this change is that you can create a new line with the same name as an existing line, which only has the effect of potentially changing the line's color, but the UI is not updated and it does not persist across saves.
Also, by not checking for `!issue`, it's possible to create a line with no color, just showing up as an invisible line.

**Review by jpwright — CHANGES_REQUESTED** [review]
Hi there, thanks again for submitting! There are a lot of really great changes in here, I like having the CD set up and some cosmetic improvements bringing things up to date a bit. A couple things before we could merge:
* Avoid changing the name "Brand New Subway" anywhere. I marked a few spots.
* I'm not sure about the changes to New Line behavior. It's not really clear to me how it allows you to make branching or looped lines.
* It would be good to avoid adding in additional `console.log` statements to this already very messy codebase ;-)
* In general you might consider submitting smaller PRs in the future, it is a bit easier to review and approve one issue at a time.

**Comment by jpwright** [comment]
> There might be one regression: the city picker doesn't work in the-singularity-labs or mine.

Yes, this is because I have my API key set up to only work for requests coming from `jpwright.github.io`. You could change the API key in your local setup if you want to mess with it.

File: js/subway-main.js
Comment (bootmii) [reply js/subway-main.js:345]: Put the if(!issue) back in in a judicious place

**Comment by bootmii** [comment]
Changes have been made. Feel free to revert the Google Analytics comment if you want.

**Comment by bootmii** [comment]
The changes to new line behavior allow the "solution" used for the A and the AirTrain to be applied elsewhere. That's what they do.

**Comment by bootmii** [comment]
Where is this API key for the city picker? I found that it uses Google Maps, but no mention of an API key in index.html (other than Google Analytics) or subway-main.js.

**Comment by jpwright** [comment]
Thanks for implementing the suggested changes. I am fine with removing Google Analytics

> The changes to new line behavior allow the "solution" used for the A and the AirTrain to be applied elsewhere. That's what they do.

Could you give me a step-by-step of how to use this on a new map to create a branching line? 

The A/AirTrain implementation is definitely a hack, but the end result is a "hidden" second A train used for the Lefferts Blvd branch, which isn't selectable in the UI, meaning you can't actually build new stations on that branch.

> Where is this API key for the city picker? I found that it uses Google Maps, but no mention of an API key in index.html (other than Google Analytics) or subway-main.js.

https://github.com/jpwright/subway/blob/master/index.html#L49

**Comment by bootmii** [comment]
...huh. What it does instead seems to be replacing the color on the map, including station pop-ups, but not in the line chooser. Maybe it's better to leave it out.

I should probably redo this whole PR and base it on your repo directly. I'll credit @pescennius and @micheletest where necessary.

**Pull request closed without merging**
